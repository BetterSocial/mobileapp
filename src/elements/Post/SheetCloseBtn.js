import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';
import {colors} from '../../utils/colors';
import {fonts} from '../../utils/fonts';
import {Button} from '../../components/Button';
import Gap from '../../components/Gap';

const SheetCloseBtn = ({backRef, goBack, continueToEdit}) => {
  return (
    <RBSheet
      ref={backRef}
      closeOnDragDown={true}
      closeOnPressMask={true}
      customStyles={{
        container: {
          borderTopRightRadius: 20,
          borderTopLeftRadius: 20,
          height: 240,
        },
        draggableIcon: {
          backgroundColor: colors.alto,
        },
      }}>
      <View style={styles.container}>
        <Text style={styles.header}>Are you sure?</Text>
        <Gap style={{height: 30}} />
        <Button onPress={continueToEdit}>
          <Text>Continue editing</Text>
        </Button>
        <Gap style={{height: 10}} />
        <Button
          onPress={goBack}
          style={{backgroundColor: colors.porcelain}}
          textStyling={{color: colors.black}}>
          <Text>Discard post</Text>
        </Button>
      </View>
    </RBSheet>
  );
};

export default SheetCloseBtn;

const styles = StyleSheet.create({
  container: {
    paddingTop: 20,
    paddingBottom: 38,
    paddingHorizontal: 20,
  },
  header: {
    color: colors.black,
    fontFamily: fonts.inter[600],
    fontSize: 18,
    fontWeight: 'bold',
  },
  containerBtn: {flexDirection: 'row'},
});
