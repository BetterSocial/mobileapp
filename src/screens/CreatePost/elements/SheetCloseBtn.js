import * as React from 'react';
import {StyleSheet, Text, View} from 'react-native';

import RBSheet from 'react-native-raw-bottom-sheet';

import {Button} from '../../../components/Button';
import Gap from '../../../components/Gap';
import {fonts} from '../../../utils/fonts';
import {COLORS} from '../../../utils/theme';

const SheetCloseBtn = ({backRef, goBack, continueToEdit}) => {
  return (
    <RBSheet
      ref={backRef}
      closeOnDragDown={true}
      closeOnPressMask={true}
      customStyles={{
        container: styles.containerSheet,
        draggableIcon: styles.draggableIcon
      }}>
      <View style={styles.container}>
        <Text style={styles.header}>Are you sure?</Text>
        <Gap style={styles.gap(30)} />
        <Button onPress={continueToEdit}>
          <Text>Continue editing</Text>
        </Button>
        <Gap style={styles.gap(10)} />
        <Button
          onPress={goBack}
          styles={{
            backgroundColor: COLORS.porcelain
          }}
          textStyling={{color: COLORS.black}}>
          <Text>Discard post</Text>
        </Button>
      </View>
    </RBSheet>
  );
};

export default SheetCloseBtn;

const styles = StyleSheet.create({
  container: {
    paddingTop: 20,
    paddingBottom: 38,
    paddingHorizontal: 20
  },
  header: {
    color: COLORS.black,
    fontFamily: fonts.inter[600],
    fontSize: 18,
    fontWeight: 'bold'
  },
  containerBtn: {flexDirection: 'row'},
  containerSheet: {
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
    height: 240
  },
  draggableIcon: {
    backgroundColor: COLORS.lightgrey
  },
  gap: (height) => ({
    height
  })
});
