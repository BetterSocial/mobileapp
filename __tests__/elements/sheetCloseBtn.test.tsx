import * as React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import SheetCloseBtn from '../../src/screens/CreatePost/elements/SheetCloseBtn';

configure({adapter: new Adapter()});
// jest.useFakeTimers();

describe('component SheetCloseBtn', () => {
  it('SheetCloseBtn snapshot', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const component = shallow(
      <SheetCloseBtn
        goBack={() => {}}
        continueToEdit={() => {}}
        backRef={myRef}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('SheetCloseBtn renders correctly', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };

    const tree = renderer
      .create(
        <SheetCloseBtn
          goBack={() => {}}
          continueToEdit={() => {}}
          backRef={myRef}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
