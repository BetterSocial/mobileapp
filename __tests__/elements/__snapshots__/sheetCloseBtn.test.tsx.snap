// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SheetCloseBtn SheetCloseBtn renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
      "landscape-left",
      "landscape-right",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#00000077",
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 240,
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ccc",
                "borderRadius": 5,
                "height": 5,
                "margin": 10,
                "width": 35,
              },
              Object {
                "backgroundColor": "#E0E0E0",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingBottom": 38,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Inter-Semi-bold",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Are you sure?
        </Text>
        <View
          style={
            Array [
              Object {
                "height": 30,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#00ADB5",
              "borderRadius": 8,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "paddingHorizontal": 25,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Inter-Semi-bold",
                "fontSize": 18,
              }
            }
          >
            <Text>
              Continue editing
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 10,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#ecf0f1",
              "borderRadius": 8,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "paddingHorizontal": 25,
              "paddingVertical": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "Inter-Semi-bold",
                "fontSize": 18,
              }
            }
          >
            <Text>
              Discard post
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`component SheetCloseBtn SheetCloseBtn snapshot 1`] = `
<RBSheet
  animationType="none"
  closeDuration={200}
  closeOnDragDown={true}
  closeOnPressBack={true}
  closeOnPressMask={true}
  customStyles={
    Object {
      "container": Object {
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 240,
      },
      "draggableIcon": Object {
        "backgroundColor": "#E0E0E0",
      },
    }
  }
  dragFromTopOnly={false}
  height={260}
  keyboardAvoidingViewEnabled={false}
  minClosingHeight={0}
  onClose={null}
  onOpen={null}
  openDuration={300}
>
  <View
    style={
      Object {
        "paddingBottom": 38,
        "paddingHorizontal": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Inter-Semi-bold",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Are you sure?
    </Text>
    <Gap
      style={
        Object {
          "height": 30,
        }
      }
    />
    <Btn
      onPress={[Function]}
    >
      <Text>
        Continue editing
      </Text>
    </Btn>
    <Gap
      style={
        Object {
          "height": 10,
        }
      }
    />
    <Btn
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#ecf0f1",
        }
      }
      textStyling={
        Object {
          "color": "#000000",
        }
      }
    >
      <Text>
        Discard post
      </Text>
    </Btn>
  </View>
</RBSheet>
`;
